lib/livebook_web/live/session_live/k8s_runtime_component.ex

```diff
@@ -0,0 +1,952 @@
+        {:noreply, save_config_secret(socket, secret, changeset)}
+
+      {:error, changeset} ->
+        {:noreply, assign_nested(socket, :save_config, changeset: changeset)}
+    end
+  end
+
+  def handle_event("load_config", %{"name" => name}, socket) do
+    secret = Enum.find(socket.assigns.hub_secrets, &(&1.name == name))
+
+    case Jason.decode(secret.value) do
+      {:ok, config_defaults} ->
+        {:noreply,
+         socket
+         |> assign(config_defaults: config_defaults)
+         |> load_config_defaults()}
+
+      {:error, _} ->
+        {:noreply, socket}
+    end
+  end
+
+  @impl true
+  def handle_async(:rbac_check, {:ok, %{errors: errors, permissions: permissions}}, socket) do
+    status = if errors === [], do: :ok, else: :errors
+    {:noreply, assign(socket, :rbac, %{status: status, errors: errors, permissions: permissions})}
+  end
+
+  def handle_async(:load_namespace_options, {:ok, [:ok, {:ok, resp}]}, socket) do
+    socket =
+      case resp do
+        %Req.Response{status: 200, body: %{"items" => resources}} ->
+          namespace_options = Enum.map(resources, & &1["metadata"]["name"])
+
+          socket
+          |> assign(:namespace_options, namespace_options)
+          |> set_namespace(List.first(namespace_options))
+          |> assign(:cluster_check, %{status: :ok, error: nil})
+
+        %Req.Response{status: _other} ->
+          # cannot list namespaces
+          socket
+          |> assign(:namespace_options, nil)
+          |> assign(:cluster_check, %{status: :ok, error: nil})
+      end
+
+    {:noreply, socket}
+  end
+
+  def handle_async(:load_namespace_options, {:ok, results}, socket) do
+    {:error, error} = List.first(results, &match?({:error, _}, &1))
+
+    socket =
+      socket
+      |> assign(:namespace_options, nil)
+      |> assign(:cluster_check, %{status: :error, error: error})
+
+    {:noreply, socket}
+  end
+
+  def handle_async(:save_config, {:ok, result}, socket) do
+    socket =
+      case result do
+        :ok ->
+          assign(socket, save_config: nil)
+
+        {:error, %Ecto.Changeset{} = changeset} ->
+          assign_nested(socket, :save_config, changeset: changeset, inflight: false)
+
+        {:transport_error, error} ->
+          assign_nested(socket, :save_config, error: error, inflight: false)
+      end
+
+    {:noreply, socket}
+  end
+
+  defp label(namespace, runtime, runtime_status) do
+    reconnecting? = reconnecting?(namespace, runtime)
+
+    case {reconnecting?, runtime_status} do
+      {true, :connected} -> "Reconnect"
+      {true, :connecting} -> "Connecting..."
+      _ -> "Connect"
+    end
+  end
+
+  defp reconnecting?(namespace, runtime) do
+    match?(%Runtime.K8s{config: %{namespace: ^namespace}}, runtime)
+  end
+
+  defp create_pvc(socket, pvc) do
+    namespace = socket.assigns.namespace
+    manifest = PVC.manifest(pvc, namespace)
+    req = socket.assigns.reqs.pvc
+
+    case Kubereq.create(req, manifest) do
```

Comment:
Let's make this async also :) See the Fly component where we use `@volume_action.inflight` to show "Creating...", etc. Basically we want to avoid blocking LV on the request.